src/toolbars/ToolBarButtons.cpp

```diff
@@ -0,0 +1,125 @@
+#include <utility>
+#include <wx/app.h>
+
+#include "ToolBarButtons.h"
+
+#include "AllThemeResources.h"
+#include "../widgets/AButton.h"
+#include "Project.h"
+
+#include "../commands/CommandContext.h"
+#include "../commands/CommandManager.h"
+#include "../commands/CommandDispatch.h"
+
+// flags so 1,2,4,8 etc.
+enum {
+   TBActTooltips = 1,
+   TBActEnableDisable = 2,
+};
+
+ToolBarButtons::ToolBarButtons(ToolBar *const parent, AudacityProject & project, ButtonList buttonList, int size, int firstButtonId)
+   : mParent(parent)
+   , mProject(project)
+   , mButtonList(std::move(buttonList))
+   , mFirstButtonId(firstButtonId)
```

Comment:
The sequence of intializers should match the sequence of member declarations, because it is the latter that really determines the sequence of execution.

It is unimportant in this case but I think this is one of the warnings we should elevate to errors in our build.  In some cases there may be surprises.